Sources/PointFree/Emails/Mailgun.swift

```diff
@@ -74,24 +82,43 @@ func mailgunSend(email: Email) -> EitherIO<Prelude.Unit, SendEmailResponse> {
     .withExcept(const(unit))
 }
 
-public func mailto<T: HasHref>(_ address: String) -> Attribute<T> {
-  return href("mailto:" + address)
-}
-
 private func attachedMailgunAuthorization(_ headers: [String: String]?) -> [String: String]? {
   let secret = Data("api:\(AppEnvironment.current.envVars.mailgun.apiKey)".utf8).base64EncodedString()
   return (headers ?? [:])
     |> key("Authorization") .~ ("Basic " + secret) // TODO: Use key path subscript
 }
 
-// TODO: move to swift-prelude
-private func compact<K, V>(_ xs: [K: V?]) -> [K: V] {
-  var result = [K: V]()
-  for (key, value) in xs {
-    if let value = value {
-      result[key] = value
-    }
-  }
-  return result
+func unsubscribeEmail(
+  fromUserId userId: Database.User.Id,
+  andNewsletter newsletter: Database.EmailSetting.Newsletter,
+  boundary: String = "--"
+  ) -> EmailAddress? {
+
+  guard let payload = encrypted(
+    text: "\(userId.unwrap.uuidString)\(boundary)\(newsletter.rawValue)",
+    secret: AppEnvironment.current.envVars.appSecret
+    ) else { return nil }
+
+  return .init(unwrap: "unsub-\(payload)@pointfree.co")
 }
 
+func userIdAndNewsletter(
+  fromUnsubscribeEmail email: EmailAddress,
+  boundary: String = "--"
+  ) -> (Database.User.Id, Database.EmailSetting.Newsletter)? {
```

Comment:
this does the opposite, generates a user id and newsletter from an unsubscribe email.

both of these functions form a `PartialIso` and we could prob bake this logic into the router (right now I do it in middleware), but I've done a few iterations on this and feel comfortable with the way it is now.